fix: Could not find an option named "dump-debug-info"

App ID: 5be7858a-e211-44f2-8120-5d21ce6bf5eb

**Description**

We get error with the Shorebird patch command (only iOS).

```
⠋ Linking AOT files...
✗ Failed to link AOT files: Exception: Failed to link: Unhandled exception:
Could not find an option named "dump-debug-info".
Usage: aot_tools link [arguments]
-h, --help                            Print this usage information.
    --base (mandatory)                Path to the base snapshot to link against.
    --patch (mandatory)               Path to the patch snapshot to link.
    --analyze-snapshot (mandatory)    Path to analyze_snapshot binary.
    --gen-snapshot (mandatory)        Path to gen_snapshot binary.
    --kernel (mandatory)              Path to the patch kernel (.dill) file.
    --output (mandatory)              Path to the output vmcode file.
    --enable-asserts                  Whether to enable asserts.
    --linker-overrides                Path to the linker overrides json file.
    --reporter                        Set how to print link results.
          [json]                      Prints the results in json format.
          [pretty] (default)          Prints the results in a human readable format.
    --redirect-to                     Redirect output to a file.
Run "aot_tools help" to see global options.
#0      Command.usageException (package:args/command_runner.dart:466:7)
#1      CommandRunner.parse (package:args/command_runner.dart:140:15)
#2      CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:122:36)
#3      new Future.sync (dart:async/future.dart:305:31)
#4      CommandRunner.run (package:args/command_runner.dart:122:14)
#5      main.<anonymous closure> (file:***home/gha/.engine_checkout/engine/src/third_party/dart/pkg/aot_tools/bin/aot_tools.dart:9:57)
#6      _rootRun (dart:async/zone.dart:1[39](https://github.com/globalmsq/msquare-market-wallet-mobile-app/actions/runs/9069983805/job/24920772732#step:11:40)9:13)
#7      _CustomZone.run (dart:async/zone.dart:1301:19)
#8      _runZoned (dart:async/zone.dart:1804:10)
#9      runZoned (dart:async/zone.dart:1747:10)
#10     LoggerOverrides.runZoned (package:aot_tools/src/logger.dart:28:18)
#11     runWithLogger (package:aot_tools/src/logger.dart:18:26)
#12     main (file:***home/gha/.engine_checkout/engine/src/third_party/dart/pkg/aot_tools/bin/aot_tools.dart:9:26)
#13     _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:295:33)
#14     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
 (0.6s)
Error: Process completed with exit code 70.
```

**Expected Behavior**

Fixing the error in the patch command.

**Additional Context**

Flutter version v3.19.5

--- a/packages/shorebird_cli/lib/src/commands/patch/ios_patcher.dart
+++ b/packages/shorebird_cli/lib/src/commands/patch/ios_patcher.dart
@@ -287,7 +287,6 @@ class IosPatcher extends Patcher {
     required File releaseArtifact,
   }) async {
     final patch = File(_aotOutputPath);
-    final dumpDebugInfo = await aotTools.isLinkDebugInfoSupported();
 
     if (!patch.existsSync()) {
       logger.err('Unable to find patch AOT file at ${patch.path}');
@@ -312,7 +311,10 @@ class IosPatcher extends Patcher {
     final linkProgress = logger.progress('Linking AOT files');
     double? linkPercentage;
     try {
-      final dumpDebugInfoDir = Directory.systemTemp.createTempSync();
+      final dumpDebugInfoDir = await aotTools.isLinkDebugInfoSupported()
+          ? Directory.systemTemp.createTempSync()
+          : null;
+
       linkPercentage = await aotTools.link(
         base: releaseArtifact.path,
         patch: patch.path,
@@ -321,10 +323,10 @@ class IosPatcher extends Patcher {
         outputPath: _vmcodeOutputPath,
         workingDirectory: buildDirectory.path,
         kernel: artifactManager.newestAppDill().path,
-        dumpDebugInfoPath: dumpDebugInfoDir.path,
+        dumpDebugInfoPath: dumpDebugInfoDir?.path,
       );
 
-      if (dumpDebugInfo) {
+      if (dumpDebugInfoDir != null) {
         final debugInfoZip = await dumpDebugInfoDir.zipToTempFile();
         debugInfoZip.copySync(p.join('build', debugInfoFile.path));
       }

--- a/packages/shorebird_cli/test/src/commands/patch/ios_patcher_test.dart
+++ b/packages/shorebird_cli/test/src/commands/patch/ios_patcher_test.dart
@@ -896,7 +896,7 @@ void main() {
                 );
               });
 
-              group('when isLinkDebugInfoSupported', () {
+              group('when isLinkDebugInfoSupported is true', () {
                 setUp(() {
                   when(
                     aotTools.isLinkDebugInfoSupported,
@@ -919,7 +919,40 @@ void main() {
                       kernel: any(named: 'kernel'),
                       outputPath: any(named: 'outputPath'),
                       workingDirectory: any(named: 'workingDirectory'),
-                      dumpDebugInfoPath: any(named: 'dumpDebugInfoPath'),
+                      dumpDebugInfoPath: any(
+                        named: 'dumpDebugInfoPath',
+                        that: isNotNull,
+                      ),
+                    ),
+                  ).called(1);
+                });
+              });
+
+              group('when isLinkDebugInfoSupported is false', () {
+                setUp(() {
+                  when(aotTools.isLinkDebugInfoSupported)
+                      .thenAnswer((_) async => false);
+                });
+
+                test('does not pass dumpDebugInfoPath to aotTools.link',
+                    () async {
+                  await runWithOverrides(
+                    () => patcher.createPatchArtifacts(
+                      appId: appId,
+                      releaseId: releaseId,
+                    ),
+                  );
+                  verify(
+                    () => aotTools.link(
+                      base: any(named: 'base'),
+                      patch: any(named: 'patch'),
+                      analyzeSnapshot: any(named: 'analyzeSnapshot'),
+                      genSnapshot: any(named: 'genSnapshot'),
+                      kernel: any(named: 'kernel'),
+                      outputPath: any(named: 'outputPath'),
+                      workingDirectory: any(named: 'workingDirectory'),
+                      // ignore: avoid_redundant_argument_values
+                      dumpDebugInfoPath: null,
                     ),
                   ).called(1);
                 });

